modules/doi/includes/admin.form.inc

```diff
@@ -30,7 +30,48 @@ function islandora_doi_admin_form(array $form, array &$form_state) {
       '#description' => t('An identifier to call yourself, for the OpenURL endpoint. To use this service you first need to register for an account here: http://www.crossref.org/requestaccount/'),
       '#default_value' => variable_get('islandora_doi_openurl_pid', 'user@example.com'),
     ),
+    'save_configuration' => array(
+      '#type' => 'submit',
+      '#name' => 'saveConfiguration',
+      '#default_value' => t('Save configuration'),
+    ),
   );
 
-  return system_settings_form($form);
```

Comment:
Any particular reason you moved away from using `system_settings_form`?

Reply:
My understanding was that having `system_settings_form($form)` at the end of the form automagically saves all the entered data to the relevant variables. I wanted to implement this check as part of a `form_validate`, so I removed that function and saved all the variables individually in a `form_submit` step.

I'm basically cargo culting from https://github.com/Islandora/islandora_usage_stats/blob/7.x/includes/admin.form.inc

Reply:
Form validation still gets triggered with `system_settings_form`.

Reply:
Are you recommending that I add `system_settings_form` back in and take the whole `form_submit` function out?

Reply:
Yep the submit handler can be removed and the form validation will still trigger.

Reply:
Just a thought: would it not be more logical to use `#element_validate` on `$form['islandora_doi_openurl_pid']`, rather than create a validator for the whole form? After all, it is only this element that is checked...

Reply:
@bryjbrown Here is an example from the core Islandora config page.
https://github.com/Islandora/islandora/blob/7.x/includes/admin.form.inc#L181-L187

Reply:
Very true @d-r-p.

Reply:
On the other hand, in view of #280, the part that checks against the default email address should only trigger in non-legacy mode, so a form validator might be the right thing, after all...